app/assets/javascripts/helpers/profile_photo_uploader.js

```diff
@@ -8,52 +8,67 @@ Diaspora.ProfilePhotoUploader.prototype = {
         emptyError: Diaspora.I18n.t("photo_uploader.empty")
       },
 
-      onSubmit: function() {
-        $("#file-upload").addClass("loading");
-        $("#profile_photo_upload").find(".avatar").addClass("loading");
-        $("#file-upload-spinner").removeClass("hidden");
-        $("#fileInfo").show();
-      },
+      callbacks: {
+        onProgress: function(id, fileName, loaded, total) {
+          var progress = Math.round(loaded / total * 100);
+          $("#fileInfo").text(fileName + " " + progress + "%");
+        },
+        onSubmit: function() {
+          $("#file-upload").addClass("loading");
+          $("#profile_photo_upload").find(".avatar").addClass("loading");
+          $("#file-upload-spinner").removeClass("hidden");
+          $("#fileInfo").show();
+        },
+        onComplete: function(id, fileName, responseJSON) {
+          $("#file-upload-spinner").addClass("hidden");
+          $("#fileInfo").text(Diaspora.I18n.t("photo_uploader.completed", {"file": fileName}));
+          $("#file-upload").removeClass("loading");
 
-      onComplete: function(_id, fileName, responseJSON) {
-        $("#file-upload-spinner").addClass("hidden");
-        $("#fileInfo").text(Diaspora.I18n.t("photo_uploader.completed", {"file": fileName}));
-        $("#file-upload").removeClass("loading");
-
-        /* flash message prompt */
-        var message = Diaspora.I18n.t("photo_uploader.looking_good");
-        if (app.flashMessages) { app.flashMessages.success(message); }
-
-        var id = responseJSON.data.photo.id;
-        var url = responseJSON.data.photo.unprocessed_image.url;
-        var oldPhoto = $("#photo_id");
-        if (oldPhoto.length === 0) {
-          $("#update_profile_form").prepend("<input type='hidden' value='" + id + "' id='photo_id' name='photo_id'/>");
-        } else {
-          oldPhoto.val(id);
-        }
+          /* flash message prompt */
+          var message = Diaspora.I18n.t("photo_uploader.looking_good");
+          if (app.flashMessages) { app.flashMessages.success(message); }
 
-        $("#profile_photo_upload").find(".avatar").removeClass("loading");
-        $("#profile_photo_upload").find(".avatar").attr("src", url);
+          id = responseJSON.data.photo.id;
+          var url = responseJSON.data.photo.unprocessed_image.url;
+          var oldPhoto = $("#photo_id");
+          if (oldPhoto.length === 0) {
+            $("#update_profile_form")
+              .prepend("<input type='hidden' value='" + id + "' id='photo_id' name='photo_id'/>");
+          } else {
+            oldPhoto.val(id);
+          }
+
+          $("#profile_photo_upload").find(".avatar").removeClass("loading");
+          $("#profile_photo_upload").find(".avatar").attr("src", url);
+          $(".avatar[data-person_id=" + gon.user.id + "]").attr("src", url);
+        },
+        onError: function(id, name, errorReason) {
+          alert(errorReason);
```

Comment:
Like I wrote above, we need the `onError` callback and it's reason now, but I will replace the alert by a flash message and prevent the relevant `success` logic when `responseJSON.data` is undefined.
For this PR I don't want to change or improve the logic of the profile_photo_uploader.js in general, because I plan to redesign the whole profile picture uploading feature.

Question: Is there a method to check, if we are in the mobile view or not?